framework/harness/mock_endpoints.go

```diff
@@ -47,6 +47,34 @@ type MockEndpoint struct {
 	closing     sync.Once
 }
 
+type MockEndpointOption helpers.ConfigOption[MockEndpoint]
+
+type mockEndpointOptionContextFn struct {
+	contextFn func(context.Context) context.Context
+}
+
+func (o mockEndpointOptionContextFn) Configure(m *MockEndpoint) error {
+	m.contextFn = o.contextFn
+	return nil
+}
+
+func MockEndpointContextFn(fn func(context.Context) context.Context) MockEndpointOption {
+	return mockEndpointOptionContextFn{fn}
+}
+
+type mockEndpointOptionDescription struct {
+	description string
```

Comment:
There was already a description field in MockEndpoint, but I was never setting it. I've added logic to set it for the various components so we'll see things like "streaming service" in the request logging.